Foundation/NSLock.swift

```diff
@@ -36,24 +43,47 @@ open class NSLock: NSObject, NSLocking {
         pthread_mutex_destroy(mutex)
         mutex.deinitialize()
         mutex.deallocate(capacity: 1)
+#if os(OSX) || os(iOS)
+        deallocateTimedLockData(timedLockData)
+#endif
     }
     
     open func lock() {
         pthread_mutex_lock(mutex)
     }
-    
+
     open func unlock() {
         pthread_mutex_unlock(mutex)
+#if os(OSX) || os(iOS)
+        if let tld = timedLockData {
+            pthread_mutex_lock(&tld.pointee.mutex)
```

Comment:
I am not fully certain this is safe since swift can potentially write into the mutex memory when it takes the address of the mutex as a stack based copy